main.go

```diff
@@ -132,6 +171,60 @@ func main() {
 	}
 }
 
+func deleteAction(ctx context.Context, clientSet client.Client, hr *fluxhelmv2beta1.HelmRelease, timeout, interval time.Duration) {
+	instance := &fluxhelmv2beta1.HelmRelease{}
+	key := client.ObjectKey{
+		Name:      hr.Name,
+		Namespace: hr.Namespace,
+	}
+	if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
+	} else if err != nil {
+		// This means that the object was not found
+		log.Errorf("Did not find the helm release: %v", hr.Name)
+	} else {
+		instance.Annotations = hr.Annotations
```

Comment:
Why are we doing this, we go and delete the object right after this?

Reply:
I added a return if the object we are trying to delete was not found. Let me know if this does not address your comment.

Reply:
What I mean is I don't understand why we are updating the spec, instance labels and annotations of the instance and then deleting it, why not just call `clientSet.Delete(ctx, hr)`?